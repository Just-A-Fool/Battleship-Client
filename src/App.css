* {
  text-align: center;
  font-family: 'Sansita', sans-serif;
  box-sizing: border-box;
}

body {
  background-color: rgb(5, 21, 46);
  color: white;
}

h1 {
  font-size: 3rem;
  font-style: oblique;
  margin: 0;
}

h2 {
  font-size: 3rem;
}

h3 {
  font-size: 2rem;
}

p {
  font-size: 1.5rem;
}
