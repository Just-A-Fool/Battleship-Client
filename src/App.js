import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
//import Header from './Components/Header/Header';

/* import route js files */
import GameroomRoute from './Routes/GameroomRoute/GameroomRoute';
import HelpRoute from './Routes/HelpRoute/HelpRoute';
import LandingRoute from './Routes/LandingRoute/LandingRoute';
import LoginRoute from './Routes/LoginRoute/LoginRoute';
import PrivateOnlyRoute from './Components/PrivateOnlyRoute/PrivateOnlyRoute';
import PublicOnlyRoute from './Components/PublicOnlyRoute/PublicOnlyRoute';
import ResultRoute from './Routes/ResultRoute/ResultRoute';
import SettingRoute from './Routes/SettingRoute/SettingRoute';
import SignupRoute from './Routes/SignupRoute/SignupRoute';
import DashboardRoute from './Routes/DashboardRoute/DashboardRoute';
import NotFoundRoute from './Routes/NotFoundRoute/NotFoundRoute';
import DeleteAccountRoute from './Routes/DeleteAccountRoute/DeleteAccountRoute';
import PasswordChangeRoute from './Routes/PasswordChangeRoute/PasswordChangeRoute';
import ResetAccountRoute from './Routes/ResetAccountRoute/ResetAccountRoute';
import UsernameChangeRoute from './Routes/UsernameChangeRoute/UsernameChangeRoute';

export default class App extends Component {
 

  render() {
    return (
      <div className="app">
        <main>
          <Switch>
            <Route
              exact
              path={'/'}
              component={LandingRoute}
            />
            <PublicOnlyRoute
              exact
              path={'/signup'}
              component={SignupRoute}
            />
            <PublicOnlyRoute
              exact
              path={'/login'}
              component={LoginRoute}
            />
            {/* <PublicOnlyRoute 
              exact
              path={'/help'}
              component={HelpRoute}
            /> */}
            <PrivateOnlyRoute
              exact
              path={'/help'}
              component={HelpRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/dashboard'}
              component={DashboardRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/setting'}
              component={SettingRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/result'}
              component={ResultRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/gameroom'}
              component={GameroomRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/deleteAccount'}
              component={DeleteAccountRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/ResetAccount'}
              component={ResetAccountRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/passwordChange'}
              component={PasswordChangeRoute}
            />
            <PrivateOnlyRoute
              exact
              path={'/UsernameChange'}
              component={UsernameChangeRoute}
            />
            
            <Route
              component={NotFoundRoute}
            />
          </Switch>
        </main>

      </div>
    );
  }
};


