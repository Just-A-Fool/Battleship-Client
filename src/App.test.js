import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

test.skip('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});


