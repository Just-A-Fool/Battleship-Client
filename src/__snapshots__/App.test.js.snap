// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders the Landing Page by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <main>
        <ErrorBoundary>
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <withRouter(Landing)
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <Route>
                  <Landing
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div
                      className="landingPage"
                    >
                      <header
                        className="landing"
                      >
                        <h1>
                          BATTLESHIP
                        </h1>
                        <nav
                          aria-label="click to view app info"
                          className="arrow-nav"
                        >
                          <div
                            className="arrow"
                            onClick={[Function]}
                          />
                        </nav>
                      </header>
                      <div
                        aria-live="polite"
                        id="app-info"
                      >
                        <p
                          aria-label="app information"
                        >
                          Welcome to Battleship! The interactive game where you can challenge opponents in the classic game of sinking ships. This application allows the user to play a live game of battleship against another user. The user will receive live updates on their opponents advancements on the game board. Additionally, user's have the ability to leave a game and return to it later.
                        </p>
                        <p
                          aria-label="app information"
                        >
                          To test out the site please create a new account, or log in with the username: TestUser1 and password: Password123. To test the live game feature you will have to log in as a different user via a separate internet browser (ie: Safari, Chrome, Firefox etc.) OR log in on a mobile device using the username: TestUser2 and password: Password123. This will give you the ability to play against yourself in the two separate browsers. Alternatively, you may give the second user login information to a friend and challenge them in a game of Battleship. Fire Away!
                        </p>
                        <ForwardRef
                          onClick={[Function]}
                        >
                          <button
                            className="Button"
                            onClick={[Function]}
                          >
                             Proceed to Login
                          </button>
                        </ForwardRef>
                      </div>
                    </div>
                  </Landing>
                </Route>
              </withRouter(Landing)>
            </Route>
          </Switch>
        </ErrorBoundary>
      </main>
    </App>
  </Router>
</BrowserRouter>
`;
